sky/clouds/cloud.py

```diff
@@ -189,6 +189,15 @@ def check_credentials(self) -> Tuple[bool, Optional[str]]:
         """
         raise NotImplementedError
 
+    def get_cloud_user_identity(self) -> Optional[str]:
+        """Returns the identity of the user on this cloud.
```

Comment:
Can we enrich this comment? Since "identity" is a term we coin. (Clouds do have this term but have different meanings).

Some ideas:
```
(Advanced) Returns the currently active identity of the user on this cloud.

The user "identity" is associated with each SkyPilot cluster he/she creates. This is used in protecting cluster operations in a multi-identity scenario, where the same user/device can switch between different cloud identities.

The choice of what constitutes an identity is up to each cloud's implementation. In general, to suffice for the above purposes, ensure that different identities imply different resources are displayed in the cloud's console.

Example identities (see cloud implementations):
  - AWS: ....
  - GCP: email address + project ID
  - Azure: email address + subscription ID
  - <new cloud>: ...


Returns:
  None iff ...; otherwise the current user identity.
```